### DISCUSSION TITLE
can't map <C-f> and <C-b> to :back and :forward respectively

### DISCUSSION THREAD

**Discussion by asherjacobson** [esm7/obsidian-vimrc-support#121] [General]
I added this to my obsidian vimrc:

exmap back obcommand app:go-back
nmap <C-b> :back
exmap forward obcommand app:go-forward
nmap <C-f> :forward

It doesn't work. Why not?

**Comment by esm7** [comment]
One thing that pops up is that the syntax of `nmap` is incorrect, you're missing a key you want to map.

**Reply by asherjacobson** [reply]
Thanks for the quick response. It looks like gitHub hid the `<C-b>` and `<C-f>` from my comment, because I didn't put it in a code block. Here is what I have in my obsidian vimrc.

```
exmap back obcommand app:go-back
nmap <C-b> :back
exmap forward obcommand app:go-forward
nmap <C-f> :forward
```

However the new behavior seems to treat ctrl b and ctrl f as previous page and next page buttons, i.e. they are now the equivalent of "alt + leftArrow" or "alt + rightArrow"; they do the same as clicking the tiny arrows in the top left corner of the obsidian app. Was this your intended behavior when you said `go back and forward with [example remappings]` at the top of the ReadMe? Obviously, it is different from the ctrl o and ctrl i behavior in vim.

**Comment by esm7** [comment]
Yeah, that was the intention. A more comprehensive implementation of a jumplist is surely something that many users will find beneficial, you can ask for it [here](https://github.com/replit/codemirror-vim).

**Reply by asherjacobson** [reply]
Thanks for the suggestion I'm confused though. Does Codemirror relate to/work with obsidian somehow?

**Comment by esm7** [comment]
CodeMirror is the editor component that Obsidian uses, and that's where the Vim support is implemented (not by Obsidian itself).

**Reply by asherjacobson** [reply]
Thanks for clarifying!

**Reply by asherjacobson** [reply]
It appears, FWIW, that ctrl o and ctrl i jumplist does work as expected if default ctrl o shortcut is removed...

**Reply by asherjacobson** [reply]
so should just be able to do `nmap <C-b> <C-o>` and same for F/I

**Comment by nesanakin** [comment]
@asherjacobson Does the <C-o> and <C-i> jumplist functionality still work for you in current Obsidian versions? I haven't been able to get it to work on the MacOS version. There are no existing shortcuts that conflict.
